lib/storage/fs/index.js

```diff
@@ -22,7 +22,11 @@ p.fetch = function(options, originalPath, stepsHash, cb) {
   fs.readFile(filename + '.json', 'utf8', function(err, data) {
     var info = { path: originalPath, stepsHash: stepsHash };
     if (data) {
-      info = _.merge(info, JSON.parse(data.toString()));
+      try {
+        info = _.merge(info, JSON.parse(data.toString()));
+      catch (err) {
+        fs.unlink(filename + '.json', function(err){});
```

Comment:
Not sure removing the file is necessary. Instead you should `return void cb(new Error('file not found'));` so everything will be overwritten appropriately.